lib/active_model/serializer/adapter.rb

```diff
@@ -41,8 +41,12 @@ def adapters
         # @param [Class] klass - adapter class itself
         # @example
         #     AMS::Adapter.register(:my_adapter, MyAdapter)
+        # @note The registered name strips out 'ActiveModel::Serializer::Adapter::'
+        #   so that registering 'ActiveModel::Serializer::Adapter::Json' and
+        #   'Json' will both register as 'json'.
         def register(name, klass)
-          adapter_map.update(name.to_s.underscore => klass)
+          name = name.to_s.remove('ActiveModel::Serializer::Adapter::'.freeze)
```

Comment:
why freeze the string? it's not gonna be used again? or does ruby cache it for the next time register is called? (cause that would be cool)
Also, if that's the case, should the string move to a class constant? like `ADAPTER_PREFIX` or something?